src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -0,0 +1,574 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ExtensionPalette.h"
+#include "../../types/inc/utils.hpp"
+#include "LibraryResources.h"
+
+#include "ExtensionPalette.g.cpp"
+#include "ChatMessage.g.cpp"
+#include "GroupedChatMessages.g.cpp"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::System;
+namespace WWH = ::winrt::Windows::Web::Http;
+namespace WSS = ::winrt::Windows::Storage::Streams;
+namespace WDJ = ::winrt::Windows::Data::Json;
+
+static constexpr std::wstring_view acceptedModel{ L"gpt-35-turbo" };
+static constexpr std::wstring_view acceptedSeverityLevel{ L"safe" };
+
+const std::wregex azureOpenAIEndpointRegex{ LR"(^https.*openai\.azure\.com)" };
```

Comment:
expensive and incorrect -- this will match `https://foo.com.bar.baz/openai.azure.com/blah`

Reply:
"byollm" amirite 🙃